src/fides/api/ops/service/connectors/consent_email_connector.py

```diff
@@ -0,0 +1,170 @@
+from fides.api.ops.schemas.redis_cache import Identity
+from fides.api.ops.service.connectors.base_connector import LimitedConnector
+from fides.api.ops.service.messaging.message_dispatch_service import dispatch_message
+from fides.core.config import get_config
+
+CONFIG = get_config()
+
+CONSENT_EMAIL_CONNECTOR_TYPES = [ConnectionType.sovrn]
+
+
+class GenericEmailConsentConnector(LimitedConnector[None]):
+    """Generic Email Consent Connector that can be overridden for specific vendors"""
+
+    @property
+    def identities_for_test_email(self) -> Dict[str, Any]:
+        """The mock user identities that are sent in the test
+        email to ensure the connector is working"""
+        return {"email": "test_email@example.com"}
+
+    @property
+    def required_identities(self) -> List[str]:
+        """Returns the identity types we need to supply to the third party for this connector"""
+        config = ExtendedConsentEmailSchema(**self.configuration.secrets or {})
+        return get_identity_types_for_connector(config)
+
+    def test_connection(self) -> Optional[ConnectionTestStatus]:
+        """
+        Sends an email to the "test_email" configured, just to establish
+        that the email workflow is working.
+        """
+        config = ExtendedConsentEmailSchema(**self.configuration.secrets or {})
+        try:
+            if not config.test_email_address:
+                raise MessageDispatchException(
+                    f"Cannot test connection. No test email defined for {self.configuration.name}"
+                )
+
+            logger.info("Starting test connection to {}", self.configuration.key)
+
+            # synchronous since failure to send is considered a connection test failure
+            send_single_consent_email(
+                db=Session.object_session(self.configuration),
+                subject_email=config.test_email_address,
+                subject_name=config.third_party_vendor_name,
+                required_identities=self.required_identities,
+                user_consent_preferences=[
+                    ConsentPreferencesByUser(
+                        identities=self.identities_for_test_email,
+                        consent_preferences=[
+                            Consent(data_use="advertising", opt_in=False),
+                            Consent(data_use="improve", opt_in=True),
+                        ],
+                    )
+                ],
+                test_mode=True,
+            )
+
+        except MessageDispatchException as exc:
+            logger.info("Email consent connector test failed with exception {}", exc)
+            return ConnectionTestStatus.failed
+        return ConnectionTestStatus.succeeded
+
+
+class SovrnConsentConnector(GenericEmailConsentConnector):
+    """SovrnConsentConnector - only need to override the details for the test email."""
+
+    @property
+    def identities_for_test_email(self) -> Dict[str, Any]:
```

Comment:
oh never mind, I guess what I commented on above just gets overwritten here?

Reply:
yes exactly that one above is for a generic connector and this one is for sovrn specifically